integration_tests/testsuite/deploy_app.py

```diff
@@ -76,17 +86,62 @@ def _record_latency_to_bigquery(deploy_latency, language, is_xrt):
 def deploy_app_and_record_latency(appdir, language, is_xrt):
     start_time = time.time()
 
-    version = deploy_app(None, None, appdir, None)
+    version, url = deploy_app(None, None, appdir, None, 'GAE')
 
     # Latency is in seconds round up to 2 decimals
     deploy_latency = round(time.time() - start_time, 2)
 
     # Store the deploy latency data to bigquery
     _record_latency_to_bigquery(deploy_latency, language, is_xrt)
-    return version
+    return version, url
+
+
+def deploy_app_gae(yaml):
+    deployed_version = test_util.generate_version()
+
+    # TODO: once sdk driver is published, use it here
+    deploy_command = ['gcloud', 'app', 'deploy', '--no-promote',
+                      '--version', deployed_version, '-q']
+    if yaml:
+        logging.info(yaml)
+        deploy_command.append(yaml)
+
+    test_util.execute_command(deploy_command, True)
+    return (deployed_version,
+            test_util.retrieve_url_for_version(deployed_version))
+
+
+def deploy_app_gke(yaml):
+    logging.debug('Starting deploy to GKE')
+
+    image_name = test_util.generate_gke_image_name()
+    service_name = test_util.generate_gke_service_name()
+    namespace = test_util.generate_namespace()
+
+    build_command = ['docker', 'build', '-t', image_name, '.']
```

Comment:
I guess it comes down you where you're planning to run this. Do you have docker installed in all the environments that will execute the tests?